enable ROCm as the compile target

han-minhee opened the pull request:
Hi, thank you for the great work!
I'm now using ROCm system, so I wanted to add it to the compile target.
It's just a few lines of additions that make "WITHROCM" works in a similar way to "WITHCUDA."
Could you check it if it would be appropriate to merge this to the main branch?

bashbaug reviewed (commented):
I don't have easy access to a ROCm system to test but this looks reasonable to me.

Would you mind updating the README to describe this new CMake option also?  Thanks!

han-minhee commented:
I added instructions on the last part of README.md and modified the CMakeLists.txt for Ch17 to disable `fig_17_31_inter_kernel_pipe` when using ROCm devices too

bashbaug reviewed (approved):
Thanks, LGTM!
